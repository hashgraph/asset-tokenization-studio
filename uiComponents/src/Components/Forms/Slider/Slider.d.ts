import type { sliderAnatomy as ChakraSliderParts } from "@chakra-ui/anatomy";
import { type SliderProps as ChakraSliderProps } from "@chakra-ui/slider";
import type { ComponentWithAs } from "@chakra-ui/system";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
export declare const sliderPartsList: typeof ChakraSliderParts.keys;
type Parts = typeof sliderPartsList;
export interface SliderThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface SliderProps extends Omit<ChakraSliderProps, "children"> {
}
export declare const Slider: ComponentWithAs<"div", SliderProps>;
export {};
