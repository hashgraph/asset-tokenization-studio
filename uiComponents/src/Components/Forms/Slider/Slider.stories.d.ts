import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Slider").SliderProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Horizontal: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Slider").SliderProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Vertical: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./Slider").SliderProps, import("@chakra-ui/system/dist/system.types").As>>;
