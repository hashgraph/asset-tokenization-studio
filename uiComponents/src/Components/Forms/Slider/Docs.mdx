import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as SliderStories from "./Slider.stories";

<Meta of={SliderStories} />

<Title />

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Slider.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Slider: ConfigSlider,
  },
};
```

```tsx dark
const ConfigSlider: SliderThemeConfiguration = {
  parts: sliderPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"filledTrack" | "track" | "container" | "thumb"` (`sliderPartsList`)

## Props

<ArgTypes of={SliderStories} />

## Stories

<Stories title="" />
