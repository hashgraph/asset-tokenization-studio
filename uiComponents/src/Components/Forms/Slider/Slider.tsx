import type { sliderAnatomy as ChakraSliderParts } from "@chakra-ui/anatomy";
import {
  Slider as ChakraSlider,
  SliderFilledTrack as ChakraSliderFilledTrack,
  SliderThumb as ChakraSliderThumb,
  SliderTrack as ChakraSliderTrack,
  type SliderProps as ChakraSliderProps,
} from "@chakra-ui/slider";
import type { ComponentWithAs } from "@chakra-ui/system";
import { forwardRef } from "@chakra-ui/system";
import React from "react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";

export const sliderPartsList: typeof ChakraSliderParts.keys = [
  "container",
  "track",
  "filledTrack",
  "thumb",
];

type Parts = typeof sliderPartsList;
export interface SliderThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export interface SliderProps extends Omit<ChakraSliderProps, "children"> {}

export const Slider: ComponentWithAs<"div", SliderProps> = forwardRef<
  SliderProps,
  "div"
>(({ ...props }: SliderProps, ref) => {
  return (
    <ChakraSlider ref={ref} {...props}>
      <ChakraSliderTrack>
        <ChakraSliderFilledTrack />
      </ChakraSliderTrack>
      <ChakraSliderThumb />
    </ChakraSlider>
  );
});
