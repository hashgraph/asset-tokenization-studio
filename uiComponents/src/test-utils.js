import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
