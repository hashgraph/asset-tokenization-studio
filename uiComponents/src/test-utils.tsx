import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
