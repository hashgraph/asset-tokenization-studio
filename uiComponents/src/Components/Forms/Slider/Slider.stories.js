var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Slider } from "./Slider";
var meta = {
    title: "Design System/Forms/Slider",
    component: Slider,
    argTypes: {
        orientation: {
            control: { type: "radio" },
            options: ["horizontal", "vertical"]
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A9914"
        }
    },
    args: {
        defaultValue: 30
    }
};
export default meta;
var Template = function (args) { return React.createElement(Slider, __assign({}, args)); };
export var Horizontal = Template.bind({});
Horizontal.args = {
    orientation: "horizontal",
    minW: 50,
    maxW: 200
};
export var Vertical = Template.bind({});
Vertical.args = {
    orientation: "vertical",
    minH: 32,
    maxH: 200
};
