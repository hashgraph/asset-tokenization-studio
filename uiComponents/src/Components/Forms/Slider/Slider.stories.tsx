import type { Meta, StoryFn } from "@storybook/react";
import React from "react";

import { Slider } from "./Slider";

const meta = {
  title: "Design System/Forms/Slider",
  component: Slider,
  argTypes: {
    orientation: {
      control: { type: "radio" },
      options: ["horizontal", "vertical"],
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A9914",
    },
  },
  args: {
    defaultValue: 30,
  },
} as Meta<typeof Slider>;
export default meta;

const Template: StoryFn<typeof Slider> = (args) => <Slider {...args} />;

export const Horizontal = Template.bind({});

Horizontal.args = {
  orientation: "horizontal",
  minW: 50,
  maxW: 200,
};

export const Vertical = Template.bind({});
Vertical.args = {
  orientation: "vertical",
  minH: 32,
  maxH: 200,
};
