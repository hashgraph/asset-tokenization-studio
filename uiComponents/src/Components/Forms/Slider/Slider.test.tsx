import { render } from "@/test-utils";
import React from "react";
import type { SliderProps } from "./Slider";
import { Slider } from "./Slider";

//@ts-ignore
const { ResizeObserver } = window;

beforeEach(() => {
  //@ts-ignore
  delete window.ResizeObserver;
  window.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
  }));
});

afterEach(() => {
  window.ResizeObserver = ResizeObserver;
  jest.restoreAllMocks();
});

describe(`<Slider />`, () => {
  const defaultProps: SliderProps = {
    defaultValue: 30,
  };

  const factoryComponent = (props: SliderProps = defaultProps) =>
    render(<Slider {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows the default value", () => {
    const defaultValue = 70;
    const component = factoryComponent({ defaultValue });
    const { container } = component;
    expect(container.querySelector("input")?.value).toBe(`${defaultValue}`);
    expect(component.asFragment()).toMatchSnapshot("with default value");
  });

  test("shows horizontal slider", () => {
    const component = factoryComponent({ orientation: "horizontal" });
    expect(component.getByRole("slider")).toBeInTheDocument();
    expect(component.getByRole("slider").getAttribute("aria-orientation")).toBe(
      "horizontal"
    );
    expect(component.asFragment()).toMatchSnapshot("horizontal slider");
  });

  test("shows vertical slider", () => {
    const component = factoryComponent({ orientation: "vertical" });
    expect(component.getByRole("slider")).toBeInTheDocument();
    expect(component.getByRole("slider").getAttribute("aria-orientation")).toBe(
      "vertical"
    );
    expect(component.asFragment()).toMatchSnapshot("vertical slider");
  });
});
