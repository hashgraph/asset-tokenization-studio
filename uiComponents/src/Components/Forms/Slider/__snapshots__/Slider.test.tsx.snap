// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-slider css-10gqts"
    style="position: relative; user-select: none; outline: 0; padding-top: 0px; padding-bottom: 0px;"
    tabindex="-1"
  >
    <div
      class="chakra-slider__track css-k3nf9g"
      id="slider-track-:r0:"
      style="position: absolute; top: 50%; transform: translateY(-50%); width: 100%;"
    >
      <div
        class="chakra-slider__filled-track css-q6qamm"
        style="position: absolute; top: 50%; transform: translateY(-50%); width: 30%; left: 0%;"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="30"
      class="chakra-slider__thumb css-1ky2tds"
      id="slider-thumb-:r0:"
      role="slider"
      style="position: absolute; user-select: none; left: calc(30% - 0px);"
      tabindex="0"
    />
    <input
      type="hidden"
      value="30"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Slider /> shows horizontal slider: horizontal slider 1`] = `
<DocumentFragment>
  <div
    class="chakra-slider css-10gqts"
    style="position: relative; user-select: none; outline: 0; padding-top: 0px; padding-bottom: 0px;"
    tabindex="-1"
  >
    <div
      class="chakra-slider__track css-k3nf9g"
      id="slider-track-:r2:"
      style="position: absolute; top: 50%; transform: translateY(-50%); width: 100%;"
    >
      <div
        class="chakra-slider__filled-track css-q6qamm"
        style="position: absolute; top: 50%; transform: translateY(-50%); width: 50%; left: 0%;"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="chakra-slider__thumb css-1ky2tds"
      id="slider-thumb-:r2:"
      role="slider"
      style="position: absolute; user-select: none; left: calc(50% - 0px);"
      tabindex="0"
    />
    <input
      type="hidden"
      value="50"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Slider /> shows the default value: with default value 1`] = `
<DocumentFragment>
  <div
    class="chakra-slider css-10gqts"
    style="position: relative; user-select: none; outline: 0; padding-top: 0px; padding-bottom: 0px;"
    tabindex="-1"
  >
    <div
      class="chakra-slider__track css-k3nf9g"
      id="slider-track-:r1:"
      style="position: absolute; top: 50%; transform: translateY(-50%); width: 100%;"
    >
      <div
        class="chakra-slider__filled-track css-q6qamm"
        style="position: absolute; top: 50%; transform: translateY(-50%); width: 70%; left: 0%;"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="70"
      class="chakra-slider__thumb css-1ky2tds"
      id="slider-thumb-:r1:"
      role="slider"
      style="position: absolute; user-select: none; left: calc(70% - 0px);"
      tabindex="0"
    />
    <input
      type="hidden"
      value="70"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Slider /> shows vertical slider: vertical slider 1`] = `
<DocumentFragment>
  <div
    class="chakra-slider css-om0rg5"
    style="position: relative; user-select: none; outline: 0; padding-left: 0px; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="chakra-slider__track css-1i926hj"
      id="slider-track-:r3:"
      style="position: absolute; left: 50%; transform: translateX(-50%); height: 100%;"
    >
      <div
        class="chakra-slider__filled-track css-q6qamm"
        style="position: absolute; left: 50%; transform: translateX(-50%); height: 50%; bottom: 0%;"
      />
    </div>
    <div
      aria-orientation="vertical"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="chakra-slider__thumb css-1wo4imd"
      id="slider-thumb-:r3:"
      role="slider"
      style="position: absolute; user-select: none; bottom: calc(50% - 0px);"
      tabindex="0"
    />
    <input
      type="hidden"
      value="50"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
